modules/control-utility/src/main/java/org/apache/ignite/internal/commandline/DeclarativeCommandAdapter.java

```diff
@@ -129,25 +128,27 @@ public DeclarativeCommandAdapter(org.apache.ignite.internal.management.api.Comma
             null
         );
 
+        Consumer<Field> namedArgCb =
+            fld -> namedArgs.add(toArg.apply(fld, fld.getAnnotation(Argument.class).optional()));
+
+        Consumer<Field> positionalArgCb = fld -> positionalArgs.add(new CLIArgument<>(
+            fld.getName(),
+            null,
+            fld.getAnnotation(Argument.class).optional(),
+            fld.getType(),
+            null
+        ));
+
         visitCommandParams(
             cmd0.argClass(),
-            fld -> positionalArgs.add(new CLIArgument<>(
-                fld.getName(),
-                null,
-                fld.getAnnotation(Argument.class).optional(),
-                fld.getType(),
-                null
-            )),
-            fld -> namedArgs.add(toArg.apply(fld, fld.getAnnotation(Argument.class).optional())),
-            (optionals, flds) -> {
-                List<CLIArgument<?>> oneOfArg = flds.stream().map(
-                    fld -> toArg.apply(fld, fld.getAnnotation(Argument.class).optional())
-                ).collect(Collectors.toList());
-
-                oneOfArgs.add(F.t(optionals, oneOfArg));
-
-                flds.forEach(fld -> namedArgs.add(toArg.apply(fld, true)));
-            }
+            positionalArgCb,
+            namedArgCb,
+            (argGrp, flds) -> flds.forEach(fld -> {
```

Comment:
Is there any reason to leave this callback as is?

Reply:
Extracted to variable